bloodview/src/device.c

```diff
@@ -412,6 +414,10 @@ static void *device__thread(void *ctx)
 			/* Awaiting response to sent message,
 			 * or running an acquisition. */
 			device__thread_receive_msg(&sent_type);
+		} else {
+			/* When we're not waiting for messages, don't
+			 * thrash the device thread main loop. */
+			usleep(10 * 1000);
```

Comment:
It think you can use sleep(0) to concede timeslice without needing to give a timeout time for situations like this, not sure if useful though.

Reply:
OK, cool.  I'll leave it as 10ms for now, because we don't really need frequent polling here.